src/main/java/chess/domain/position/Position.java

```diff
@@ -0,0 +1,69 @@
+package chess.domain.position;
+
+import java.util.Objects;
+
+public class Position {
+    private final File file;
+    private final Rank rank;
+
+    private Position(File file, Rank rank) {
+        this.file = file;
+        this.rank = rank;
+    }
+
+    public static Position from(String position) {
+        File file = File.valueOf(position.substring(0, 1));
+        Rank rank = Rank.valueOf(position.substring(1));
+
+        return new Position(file, rank);
+    }
+
+    public Position move(int fileDirection, int rankDirection) {
+        return new Position(file.update(fileDirection), rank.update(rankDirection));
+    }
+
+    public int calculateFileDistance(Position target) {
+        return Math.abs(file.subtractFile(target.file));
+    }
+
+    public int calculateRankDistance(Position target) {
+        return Math.abs(subtractRanks(target));
+    }
+
+    public int subtractRanks(Position target) {
+        return rank.subtractRank(target.rank);
+    }
+
+    public boolean isSameRank(Position target) {
+        return this.rank == target.rank;
+    }
+
+    public boolean isSameFile(Position target) {
+        return this.file == target.file;
+    }
```

Comment:
```suggestion
    public boolean isSameRank(Position target) {
        return rank == target.rank;
    }

    public boolean isSameFile(Position target) {
        return file == target.file;
    }
```
this를 사용하는 초롱의 기준은 무엇인가요?

Reply:
보통은 매개변수와 멤버변수의 이름이 동일할 때 사용을 하는 편입니다.
하지만 아직은 의식하고 사용하지 않으면 썼다가 안썼다가 왔다갔다하는 것 같습니다😅

Reply:
미션을 진행하면서 초롱만의 기준을 세우시는걸 추천드려요 :)